altizon/io/datonis/edge/edge_gateway_mqtt.py

```diff
@@ -102,10 +126,9 @@ def connect(self):
         if retval == 0:
             self.ack_lock = threading.Condition()
             self.mqtt_client.loop_start()
+            self.state = CONNECTING
             # Start a new thread for instruction execution
             thread.start_new_thread(instruction_worker, ('instruction-worker', self))
-            self.subscribe_for_instructions()
```

Comment:
Were we subscribing for instructions/acks twice? Once here and once in on_connect()?

Reply:
Yes, on_connect callback subscribes for instruction after connection.